Switch webpushd over to using generated IPC encoder/decoder

beidson opened the pull request:
#### 70dba30bd03c14d661efe268d25e418018fdc9d9
<pre>
Switch webpushd over to using generated IPC encoder/decoder
<a href="https://bugs.webkit.org/show_bug.cgi?id=261361">https://bugs.webkit.org/show_bug.cgi?id=261361</a>
rdar://114886414

Reviewed by Chris Dumez.

All of these hand rolled and hand maintained messages and related macros and templates are... fragile.
Let&apos;s generate them, greatly improving the ability to hack on webpushd related stuff quickly.

This patch:
- Generates messages/serializers for PushClientConnection stuff
- Removes all of the raw xpc messaging to webpushd
- Exposes these messages and ArgumentCoders to TestWebKitAPI and webpushtool
- Adapts TestWebKitAPI and webpushtool to these changes

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources-output.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/NetworkProcess/Notifications/Cocoa/WebPushDaemonConnectionCocoa.mm:
(WebKit::WebPushD::Connection::newConnectionWasInitialized const):
(WebKit::WebPushD::messageDictionaryFromEncoder):
(WebKit::WebPushD::Connection::performSendWithoutUsingIPCConnection const):
(WebKit::WebPushD::Connection::performSendWithAsyncReplyWithoutUsingIPCConnection const):
(WebKit::WebPushD::addVersionAndEncodedMessageToDictionary): Deleted.
(WebKit::WebPushD::handleWebPushDaemonMessage): Deleted.
(WebKit::WebPushD::Connection::connectionReceivedEvent): Deleted.
(WebKit::WebPushD::Connection::dictionaryFromMessage const): Deleted.
* Source/WebKit/NetworkProcess/Notifications/NetworkNotificationManager.cpp:
(WebKit::NetworkNotificationManager::setPushAndNotificationsEnabledForOrigin):
(WebKit::NetworkNotificationManager::deletePushAndNotificationRegistration):
(WebKit::NetworkNotificationManager::getPendingPushMessages):
(WebKit::NetworkNotificationManager::subscribeToPushService):
(WebKit::NetworkNotificationManager::unsubscribeFromPushService):
(WebKit::NetworkNotificationManager::getPushSubscription):
(WebKit::NetworkNotificationManager::getPushPermissionState):
(WebKit::NetworkNotificationManager::incrementSilentPushCount):
(WebKit::NetworkNotificationManager::removeAllPushSubscriptions):
(WebKit::NetworkNotificationManager::removePushSubscriptionsForOrigin):
(WebKit::NetworkNotificationManager::sendMessage const): Deleted.
(WebKit::ReplyCaller&lt;&gt;::callReply): Deleted.
(WebKit::ReplyCaller&lt;String&gt;::callReply): Deleted.
(): Deleted.
(WebKit::ReplyCaller&lt;bool&gt;::callReply): Deleted.
(WebKit::ReplyCaller&lt;unsigned&gt;::callReply): Deleted.
(WebKit::ReplyCaller&lt;Vector&lt;String&gt;::callReply): Deleted.
(WebKit::ReplyCaller&lt;Vector&lt;WebPushMessage&gt;::callReply): Deleted.
(WebKit::NetworkNotificationManager::sendMessageWithReply const): Deleted.
* Source/WebKit/NetworkProcess/Notifications/NetworkNotificationManager.h:
* Source/WebKit/NetworkProcess/Notifications/WebPushDaemonConnection.h:
* Source/WebKit/Platform/IPC/DaemonCoders.cpp:
* Source/WebKit/Platform/IPC/MessageSender.cpp:
(IPC::MessageSender::performSendWithoutUsingIPCConnection const):
(IPC::MessageSender::performSendWithAsyncReplyWithoutUsingIPCConnection const):
(IPC::MessageSender::performSendWithoutUsingIPCConnection): Deleted.
(IPC::MessageSender::performSendWithAsyncReplyWithoutUsingIPCConnection): Deleted.
* Source/WebKit/Platform/IPC/MessageSender.h:
* Source/WebKit/Platform/IPC/MessageSenderInlines.h:
(IPC::MessageSender::sendWithoutUsingIPCConnection const):
(IPC::MessageSender::sendWithAsyncReplyWithoutUsingIPCConnection const):
(IPC::MessageSender::sendWithoutUsingIPCConnection): Deleted.
(IPC::MessageSender::sendWithAsyncReplyWithoutUsingIPCConnection): Deleted.
* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
(generate_message_handler):
* Source/WebKit/Scripts/webkit/tests/TestWithoutUsingIPCConnectionMessageReceiver.cpp:
(WebKit::TestWithoutUsingIPCConnection::didReceiveMessageWithReplyHandler):
* Source/WebKit/Shared/Daemon/DaemonUtilities.h:
* Source/WebKit/Shared/Daemon/DaemonUtilities.mm:
(WebKit::encoderToXPCData):
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebPushDaemonConnectionConfiguration.h:
(WebKit::WebPushD::WebPushDaemonConnectionConfiguration::encode const): Deleted.
(WebKit::WebPushD::WebPushDaemonConnectionConfiguration::decode): Deleted.
* Source/WebKit/Shared/WebPushDaemonConnectionConfiguration.serialization.in:
* Source/WebKit/Shared/WebPushDaemonConstants.h:
(WebKit::WebPushD::messageTypeSendsReply): Deleted.
(WebKit::WebPushD::daemonMessageTypeSendsReply): Deleted.
* Source/WebKit/Sources.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/webpushd/PushClientConnection.h:
* Source/WebKit/webpushd/PushClientConnection.messages.in: Added.
* Source/WebKit/webpushd/PushClientConnection.mm:
(WebPushD::PushClientConnection::updateConnectionConfiguration):
(WebPushD::PushClientConnection::getPushTopicsForTesting):
(WebPushD::PushClientConnection::sendDebugMessage):
(WebPushD::PushClientConnection::setPushAndNotificationsEnabledForOrigin):
(WebPushD::PushClientConnection::deletePushAndNotificationRegistration):
(WebPushD::PushClientConnection::injectPushMessageForTesting):
(WebPushD::PushClientConnection::injectEncryptedPushMessageForTesting):
(WebPushD::PushClientConnection::getPendingPushMessages):
(WebPushD::PushClientConnection::subscribeToPushService):
(WebPushD::PushClientConnection::unsubscribeFromPushService):
(WebPushD::PushClientConnection::getPushSubscription):
(WebPushD::PushClientConnection::incrementSilentPushCount):
(WebPushD::PushClientConnection::removeAllPushSubscriptions):
(WebPushD::PushClientConnection::removePushSubscriptionsForOrigin):
(WebPushD::PushClientConnection::setPublicTokenForTesting):
(WebPushD::PushClientConnection::getPushPermissionState):
(WebPushD::PushClientConnection::sendDaemonMessage const): Deleted.
* Source/WebKit/webpushd/WebPushDaemon.h:
* Source/WebKit/webpushd/WebPushDaemon.mm:
(WebPushD::tryCloseRequestConnection):
(WebPushD::WebPushDaemon::connectionEventHandler):
(WebPushD::WebPushDaemon::setPushAndNotificationsEnabledForOrigin):
(WebPushD::WebPushDaemon::deletePushAndNotificationRegistration):
(WebPushD::WebPushDaemon::injectPushMessageForTesting):
(WebPushD::WebPushDaemon::injectEncryptedPushMessageForTesting):
(WebPushD::WebPushDaemon::getPendingPushMessages):
(WebPushD::WebPushDaemon::getPushTopicsForTesting):
(WebPushD::WebPushDaemon::subscribeToPushService):
(WebPushD::WebPushDaemon::unsubscribeFromPushService):
(WebPushD::WebPushDaemon::getPushSubscription):
(WebPushD::WebPushDaemon::incrementSilentPushCount):
(WebPushD::WebPushDaemon::removeAllPushSubscriptions):
(WebPushD::WebPushDaemon::removePushSubscriptionsForOrigin):
(WebPushD::WebPushDaemon::setPublicTokenForTesting):
(WebPushD::WebPushDaemon::canRegisterForNotifications):
(WebPushD::MessageInfo::echoTwice::encodeReply): Deleted.
(WebPushD::MessageInfo::deletePushAndNotificationRegistration::encodeReply): Deleted.
(WebPushD::MessageInfo::injectPushMessageForTesting::encodeReply): Deleted.
(WebPushD::MessageInfo::injectEncryptedPushMessageForTesting::encodeReply): Deleted.
(WebPushD::MessageInfo::getPendingPushMessages::encodeReply): Deleted.
(WebPushD::MessageInfo::subscribeToPushService::encodeReply): Deleted.
(WebPushD::MessageInfo::unsubscribeFromPushService::encodeReply): Deleted.
(WebPushD::MessageInfo::getPushSubscription::encodeReply): Deleted.
(WebPushD::MessageInfo::getPushPermissionState::encodeReply): Deleted.
(WebPushD::MessageInfo::incrementSilentPushCount::encodeReply): Deleted.
(WebPushD::MessageInfo::removeAllPushSubscriptions::encodeReply): Deleted.
(WebPushD::MessageInfo::removePushSubscriptionsForOrigin::encodeReply): Deleted.
(WebPushD::handleWebPushDMessageWithReply): Deleted.
(WebPushD::handleWebPushDMessage): Deleted.
(WebPushD::CompletionHandler&lt;void): Deleted.
(WebPushD::WebPushDaemon::decodeAndHandleRawXPCMessage): Deleted.
(WebPushD::WebPushDaemon::decodeAndHandleMessage): Deleted.
(WebPushD::WebPushDaemon::echoTwice): Deleted.
(WebPushD::WebPushDaemon::setDebugModeIsEnabled): Deleted.
(WebPushD::WebPushDaemon::updateConnectionConfiguration): Deleted.
(WebPushD::toXPCArray): Deleted.
(WebPushD::WebPushDaemon::getPushPermissionState): Deleted.
* Source/WebKit/webpushd/webpushtool/WebPushToolConnection.h:
* Source/WebKit/webpushd/webpushtool/WebPushToolConnection.mm:
(WebPushTool::Connection::sendPushMessage):
(WebPushTool::Connection::startDebugStreamAction):
(WebPushTool::Connection::sendAuditToken):
(WebPushTool::messageDictionaryFromEncoder):
(WebPushTool::Connection::performSendWithoutUsingIPCConnection const):
(WebPushTool::Connection::performSendWithAsyncReplyWithoutUsingIPCConnection const):
* Tools/TestWebKitAPI/Configurations/TestIPC.xcconfig:
* Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm:
(TestWebKitAPI::WebPushXPCConnectionMessageSender::WebPushXPCConnectionMessageSender):
(TestWebKitAPI::WebPushXPCConnectionMessageSender::incrementProtocolVersionForTesting):
(TestWebKitAPI::WebPushXPCConnectionMessageSender::messageDictionaryFromEncoder const):
(TestWebKitAPI::WebPushXPCConnectionMessageSender::performSendWithoutUsingIPCConnection const):
(TestWebKitAPI::WebPushXPCConnectionMessageSender::performSendWithAsyncReplyWithoutUsingIPCConnection const):
(TestWebKitAPI::sendConfigurationWithAuditToken):
(TestWebKitAPI::TEST):
(TestWebKitAPI::addMessageHeaders): Deleted.
(TestWebKitAPI::sendMessageToDaemon): Deleted.
(TestWebKitAPI::sendMessageToDaemonWithReplySync): Deleted.
(TestWebKitAPI::sendRawMessageToDaemonWithReplySync): Deleted.
(TestWebKitAPI::toStringVector): Deleted.
(TestWebKitAPI::encodeString): Deleted.

Canonical link: <a href="https://commits.webkit.org/267851@main">https://commits.webkit.org/267851@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e917c5d3d766bd7640edef5afc87b5b6c5cdcb27

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/17882 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/18208 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/18765 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/19709 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/16728 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/18077 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/21501 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/18362 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/18738 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/18097 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/18362 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/15545 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/20577 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/18031 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/15589 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/16298 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/22834 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/16605 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/16467 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/20700 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/17031 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/14425 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/16133 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4254 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/20491 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/16879 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/DerivedSources.make:
```diff
@@ -315,7 +316,8 @@ MESSAGE_RECEIVERS = \
 	GPUProcess/media/RemoteMediaSourceProxy \
 	GPUProcess/media/RemoteRemoteCommandListenerProxy \
-	GPUProcess/media/RemoteSourceBufferProxy \
+    GPUProcess/media/RemoteSourceBufferProxy \
```
indentation issue.

cdumez commented on Source/WebKit/webpushd/webpushtool/WebPushToolConnection.mm:
```diff
@@ -136,35 +139,18 @@ static mach_port_t maybeConnectToService(const char* serviceName)
-    xpc_connection_send_message_with_reply(m_connection.get(), dictionary.get(), dispatch_get_main_queue(), ^(xpc_object_t resultMessage) {
-        printf("Push message injected\n");
+    sendWithAsyncReplyWithoutUsingIPCConnection(Messages::PushClientConnection::InjectPushMessageForTesting(*m_pushMessage), [shouldExitAfterInject = !m_action] (bool injected) {
+        printf("Push message injected - %i\n", injected);
```
Did you mean to keep this here?

cdumez commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm:
```diff
@@ -162,48 +163,86 @@ static void cleanUpTestWebPushD(NSURL *tempDir)
-    xpc_dictionary_set_uint64(request, WebKit::WebPushD::protocolMessageTypeKey, static_cast<uint64_t>(messageType));
-    xpc_dictionary_set_uint64(request, WebKit::WebPushD::protocolVersionKey, version);
-}
+class WebPushXPCConnectionMessageSender : public IPC::MessageSender {
```
Can we mark the class as final?

cdumez commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm:
```diff
@@ -162,48 +163,86 @@ static void cleanUpTestWebPushD(NSURL *tempDir)
+    IPC::Connection* messageSenderConnection() const final { return nullptr; }
+    uint64_t messageSenderDestinationID() const final { return 0; }
+    OSObjectPtr<xpc_connection_t> m_connection;
+    bool m_incrementProtocolVersionForTesting { false };
```
m_shouldIncrementProtocolVersionForTesting? To follow coding style?

cdumez commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm:
```diff
@@ -162,48 +163,86 @@ static void cleanUpTestWebPushD(NSURL *tempDir)
-    xpc_dictionary_set_data(request.get(), WebKit::WebPushD::protocolEncodedMessageKey, message.data(), message.size());
-    xpc_connection_send_message(connection, request.get());
-}
+    void incrementProtocolVersionForTesting() { m_incrementProtocolVersionForTesting = true; }
```
setShouldIncrementProtocolVersionForTesting() ?

cdumez commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm:
```diff
@@ -162,48 +163,86 @@ static void cleanUpTestWebPushD(NSURL *tempDir)
+
+    IPC::Connection* messageSenderConnection() const final { return nullptr; }
+    uint64_t messageSenderDestinationID() const final { return 0; }
+    OSObjectPtr<xpc_connection_t> m_connection;
```
It'd be nice to have a blank line between to separate data members from member functions.

cdumez reviewed (approved):
LGTM

beidson committed:
Switch webpushd over to using generated IPC encoder/decoder
https://bugs.webkit.org/show_bug.cgi?id=261361
rdar://114886414

Reviewed by Chris Dumez.

All of these hand rolled and hand maintained messages and related macros and templates are... fragile.
Let's generate them, greatly improving the ability to hack on webpushd related stuff quickly.

This patch:
- Generates messages/serializers for PushClientConnection stuff
- Removes all of the raw xpc messaging to webpushd
- Exposes these messages and ArgumentCoders to TestWebKitAPI and webpushtool
- Adapts TestWebKitAPI and webpushtool to these changes

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources-output.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/NetworkProcess/Notifications/Cocoa/WebPushDaemonConnectionCocoa.mm:
(WebKit::WebPushD::Connection::newConnectionWasInitialized const):
(WebKit::WebPushD::messageDictionaryFromEncoder):
(WebKit::WebPushD::Connection::performSendWithoutUsingIPCConnection const):
(WebKit::WebPushD::Connection::performSendWithAsyncReplyWithoutUsingIPCConnection const):
(WebKit::WebPushD::addVersionAndEncodedMessageToDictionary): Deleted.
(WebKit::WebPushD::handleWebPushDaemonMessage): Deleted.
(WebKit::WebPushD::Connection::connectionReceivedEvent): Deleted.
(WebKit::WebPushD::Connection::dictionaryFromMessage const): Deleted.
* Source/WebKit/NetworkProcess/Notifications/NetworkNotificationManager.cpp:
(WebKit::NetworkNotificationManager::setPushAndNotificationsEnabledForOrigin):
(WebKit::NetworkNotificationManager::deletePushAndNotificationRegistration):
(WebKit::NetworkNotificationManager::getPendingPushMessages):
(WebKit::NetworkNotificationManager::subscribeToPushService):
(WebKit::NetworkNotificationManager::unsubscribeFromPushService):
(WebKit::NetworkNotificationManager::getPushSubscription):
(WebKit::NetworkNotificationManager::getPushPermissionState):
(WebKit::NetworkNotificationManager::incrementSilentPushCount):
(WebKit::NetworkNotificationManager::removeAllPushSubscriptions):
(WebKit::NetworkNotificationManager::removePushSubscriptionsForOrigin):
(WebKit::NetworkNotificationManager::sendMessage const): Deleted.
(WebKit::ReplyCaller<>::callReply): Deleted.
(WebKit::ReplyCaller<String>::callReply): Deleted.
(): Deleted.
(WebKit::ReplyCaller<bool>::callReply): Deleted.
(WebKit::ReplyCaller<unsigned>::callReply): Deleted.
(WebKit::ReplyCaller<Vector<String>::callReply): Deleted.
(WebKit::ReplyCaller<Vector<WebPushMessage>::callReply): Deleted.
(WebKit::NetworkNotificationManager::sendMessageWithReply const): Deleted.
* Source/WebKit/NetworkProcess/Notifications/NetworkNotificationManager.h:
* Source/WebKit/NetworkProcess/Notifications/WebPushDaemonConnection.h:
* Source/WebKit/Platform/IPC/DaemonCoders.cpp:
* Source/WebKit/Platform/IPC/MessageSender.cpp:
(IPC::MessageSender::performSendWithoutUsingIPCConnection const):
(IPC::MessageSender::performSendWithAsyncReplyWithoutUsingIPCConnection const):
(IPC::MessageSender::performSendWithoutUsingIPCConnection): Deleted.
(IPC::MessageSender::performSendWithAsyncReplyWithoutUsingIPCConnection): Deleted.
* Source/WebKit/Platform/IPC/MessageSender.h:
* Source/WebKit/Platform/IPC/MessageSenderInlines.h:
(IPC::MessageSender::sendWithoutUsingIPCConnection const):
(IPC::MessageSender::sendWithAsyncReplyWithoutUsingIPCConnection const):
(IPC::MessageSender::sendWithoutUsingIPCConnection): Deleted.
(IPC::MessageSender::sendWithAsyncReplyWithoutUsingIPCConnection): Deleted.
* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
(generate_message_handler):
* Source/WebKit/Scripts/webkit/tests/TestWithoutUsingIPCConnectionMessageReceiver.cpp:
(WebKit::TestWithoutUsingIPCConnection::didReceiveMessageWithReplyHandler):
* Source/WebKit/Shared/Daemon/DaemonUtilities.h:
* Source/WebKit/Shared/Daemon/DaemonUtilities.mm:
(WebKit::encoderToXPCData):
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebPushDaemonConnectionConfiguration.h:
(WebKit::WebPushD::WebPushDaemonConnectionConfiguration::encode const): Deleted.
(WebKit::WebPushD::WebPushDaemonConnectionConfiguration::decode): Deleted.
* Source/WebKit/Shared/WebPushDaemonConnectionConfiguration.serialization.in:
* Source/WebKit/Shared/WebPushDaemonConstants.h:
(WebKit::WebPushD::messageTypeSendsReply): Deleted.
(WebKit::WebPushD::daemonMessageTypeSendsReply): Deleted.
* Source/WebKit/Sources.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/webpushd/PushClientConnection.h:
* Source/WebKit/webpushd/PushClientConnection.messages.in: Added.
* Source/WebKit/webpushd/PushClientConnection.mm:
(WebPushD::PushClientConnection::updateConnectionConfiguration):
(WebPushD::PushClientConnection::getPushTopicsForTesting):
(WebPushD::PushClientConnection::sendDebugMessage):
(WebPushD::PushClientConnection::setPushAndNotificationsEnabledForOrigin):
(WebPushD::PushClientConnection::deletePushAndNotificationRegistration):
(WebPushD::PushClientConnection::injectPushMessageForTesting):
(WebPushD::PushClientConnection::injectEncryptedPushMessageForTesting):
(WebPushD::PushClientConnection::getPendingPushMessages):
(WebPushD::PushClientConnection::subscribeToPushService):
(WebPushD::PushClientConnection::unsubscribeFromPushService):
(WebPushD::PushClientConnection::getPushSubscription):
(WebPushD::PushClientConnection::incrementSilentPushCount):
(WebPushD::PushClientConnection::removeAllPushSubscriptions):
(WebPushD::PushClientConnection::removePushSubscriptionsForOrigin):
(WebPushD::PushClientConnection::setPublicTokenForTesting):
(WebPushD::PushClientConnection::getPushPermissionState):
(WebPushD::PushClientConnection::sendDaemonMessage const): Deleted.
* Source/WebKit/webpushd/WebPushDaemon.h:
* Source/WebKit/webpushd/WebPushDaemon.mm:
(WebPushD::tryCloseRequestConnection):
(WebPushD::WebPushDaemon::connectionEventHandler):
(WebPushD::WebPushDaemon::setPushAndNotificationsEnabledForOrigin):
(WebPushD::WebPushDaemon::deletePushAndNotificationRegistration):
(WebPushD::WebPushDaemon::injectPushMessageForTesting):
(WebPushD::WebPushDaemon::injectEncryptedPushMessageForTesting):
(WebPushD::WebPushDaemon::getPendingPushMessages):
(WebPushD::WebPushDaemon::getPushTopicsForTesting):
(WebPushD::WebPushDaemon::subscribeToPushService):
(WebPushD::WebPushDaemon::unsubscribeFromPushService):
(WebPushD::WebPushDaemon::getPushSubscription):
(WebPushD::WebPushDaemon::incrementSilentPushCount):
(WebPushD::WebPushDaemon::removeAllPushSubscriptions):
(WebPushD::WebPushDaemon::removePushSubscriptionsForOrigin):
(WebPushD::WebPushDaemon::setPublicTokenForTesting):
(WebPushD::WebPushDaemon::canRegisterForNotifications):
(WebPushD::MessageInfo::echoTwice::encodeReply): Deleted.
(WebPushD::MessageInfo::deletePushAndNotificationRegistration::encodeReply): Deleted.
(WebPushD::MessageInfo::injectPushMessageForTesting::encodeReply): Deleted.
(WebPushD::MessageInfo::injectEncryptedPushMessageForTesting::encodeReply): Deleted.
(WebPushD::MessageInfo::getPendingPushMessages::encodeReply): Deleted.
(WebPushD::MessageInfo::subscribeToPushService::encodeReply): Deleted.
(WebPushD::MessageInfo::unsubscribeFromPushService::encodeReply): Deleted.
(WebPushD::MessageInfo::getPushSubscription::encodeReply): Deleted.
(WebPushD::MessageInfo::getPushPermissionState::encodeReply): Deleted.
(WebPushD::MessageInfo::incrementSilentPushCount::encodeReply): Deleted.
(WebPushD::MessageInfo::removeAllPushSubscriptions::encodeReply): Deleted.
(WebPushD::MessageInfo::removePushSubscriptionsForOrigin::encodeReply): Deleted.
(WebPushD::handleWebPushDMessageWithReply): Deleted.
(WebPushD::handleWebPushDMessage): Deleted.
(WebPushD::CompletionHandler<void): Deleted.
(WebPushD::WebPushDaemon::decodeAndHandleRawXPCMessage): Deleted.
(WebPushD::WebPushDaemon::decodeAndHandleMessage): Deleted.
(WebPushD::WebPushDaemon::echoTwice): Deleted.
(WebPushD::WebPushDaemon::setDebugModeIsEnabled): Deleted.
(WebPushD::WebPushDaemon::updateConnectionConfiguration): Deleted.
(WebPushD::toXPCArray): Deleted.
(WebPushD::WebPushDaemon::getPushPermissionState): Deleted.
* Source/WebKit/webpushd/webpushtool/WebPushToolConnection.h:
* Source/WebKit/webpushd/webpushtool/WebPushToolConnection.mm:
(WebPushTool::Connection::sendPushMessage):
(WebPushTool::Connection::startDebugStreamAction):
(WebPushTool::Connection::sendAuditToken):
(WebPushTool::messageDictionaryFromEncoder):
(WebPushTool::Connection::performSendWithoutUsingIPCConnection const):
(WebPushTool::Connection::performSendWithAsyncReplyWithoutUsingIPCConnection const):
* Tools/TestWebKitAPI/Configurations/TestIPC.xcconfig:
* Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm:
(TestWebKitAPI::WebPushXPCConnectionMessageSender::WebPushXPCConnectionMessageSender):
(TestWebKitAPI::WebPushXPCConnectionMessageSender::incrementProtocolVersionForTesting):
(TestWebKitAPI::WebPushXPCConnectionMessageSender::messageDictionaryFromEncoder const):
(TestWebKitAPI::WebPushXPCConnectionMessageSender::performSendWithoutUsingIPCConnection const):
(TestWebKitAPI::WebPushXPCConnectionMessageSender::performSendWithAsyncReplyWithoutUsingIPCConnection const):
(TestWebKitAPI::sendConfigurationWithAuditToken):
(TestWebKitAPI::TEST):
(TestWebKitAPI::addMessageHeaders): Deleted.
(TestWebKitAPI::sendMessageToDaemon): Deleted.
(TestWebKitAPI::sendMessageToDaemonWithReplySync): Deleted.
(TestWebKitAPI::sendRawMessageToDaemonWithReplySync): Deleted.
(TestWebKitAPI::toStringVector): Deleted.
(TestWebKitAPI::encodeString): Deleted.

Canonical link: https://commits.webkit.org/267851@main
